### PR TITLE
Fixissue5784 by adding checks for circuit size

### PR DESCRIPTION
**Pull request by TheGupta2012**
### Summary
Fixes issue #5784

### Details and comments
This issue was present in the `latex`, `text` and `mpl` drawers in the visualization library. Primarily added checks for the **size of the circuits** that were being passed in the drawers. The singleton registers, now, do not contain indices as was mentioned in the original issue.

Here are some screenshots of the expected results - 

```
from qiskit import * 
q2 = QuantumCircuit(3,2)
q  = QuantumCircuit(1)
q.h(0)
q.measure_all()
q2.x([0,1])
q2.cx(0,2)
q2.measure(0,1)

```

`q.draw(output='mpl',cregbundle=False,filename = "mpl_circ_1.JPG")`
`q.draw(output='mpl',filename = 'mpl_circ_2.JPG')`
`q2.draw(output='mpl',filename = 'mpl_circ_3.JPG')`

mpl_circ_1
mpl_circ_2
mpl_circ_3

`q2.draw(output='text', filename = "text_circ_3.JPG")`
`q.draw(output='text',filename = "text_circ_1.JPG")`
`q.draw(output='text',cregbundle=False,filename = "text_circ_2.JPG")`

text_circ_3
text_circ_1
text_circ_2

`q.draw(output = 'latex', cregbundle = True)`
`q.draw(ouput = 'latex', cregbundle = False)`
`q2.draw(output = 'latex')`

latex_issue_5784_1
latex_issue_5784_2
latex_issue_5784_3

### CODE REVIEW

File: qiskit/visualization/text.py [hunk 1]
```diff
@@ -776,19 +776,35 @@ def wire_names(self, with_initial_state=False):
...
+                                index=index,
+                                physical="",
+                            )
+                        )
+                    else:
+                        qubit_labels.append(
+                            label.format(
+                                name=self.bit_locations[bit]["register"].name,
+                                physical="",
+                            )
```
Comment (TharrmashasthaPV) [inline qiskit/visualization/text.py:801]: Do you think you it is possible to do something similar to what has been mentioned in comment at line 819?

**Review by TharrmashasthaPV — CHANGES_REQUESTED** [review]
Thanks for working on this @TheGupta2012 . I believe I am having some troubles with the latex drawer. For instance, take the below code
```python
qc = QuantumCircuit(1,1)
qc.h(0)
qc.draw('latex_source')
```
This gives the following output
```latex
\documentclass[border=2px]{standalone}
        
\usepackage[braket, qm]{qcircuit}
\usepackage{graphicx}

\begin{document} 
\scalebox{1.0}{
\Qcircuit @C=1.0em @R=0.2em @!R { \\
	 	\nghost{\lstick{ {q} :  } & \lstick{\lstick{ {q} :  } & \gate{\mathrm{H}} & \qw & \qw\\ 
	 	\nghost{c:} & \lstick{c:} & \lstick{/_{_{1}}} \cw & \cw & \cw\\ 
\\ }}
\end{document}
```

See that there are some closing brackets missing and there is an `\lstick` element inside another `\lstick` element. This raises error in latex. Do you mind fixing them? I also have a few pointers that I have mentioned as comments.

**Comment by enavarro51** [comment]
@TheGupta2012 Just checking to see if you are still working on this. Thanks.

**Comment by javabster** [comment]
@TheGupta2012  as we haven't heard from you I will reassign to @enavarro51 as he has asked to pick it up. If you would still like to work on it please reach out to Ed about continuing to work on it together 😄

**Comment by TheGupta2012** [comment]
> @TheGupta2012 as we haven't heard from you I will reassign to @enavarro51 as he has asked to pick it up. If you would still like to work on it please reach out to Ed about continuing to work on it together 😄

I am extremely sorry @javabster and @enavarro51, I haven't been able to take out the time for the past three weeks due to my college exams and tests. @enavarro51 you may continue to work on the fix and as soon as I am able to, I shall also contribute to the same.  
Again, really sorry for the delay..

**Comment by enavarro51** [comment]
@TheGupta2012 Thanks for the reply. No problem. We know everyone is busy with other things. I will start working on this in the next few days. I think I already have the items @TharrmashasthaPV reviewed above taken care of. The big task is fixing all the text drawer tests. There have been some changes merged into main that require some fixing. 

I'll just move along with it and if you have some time just let me know. I'll invite you as a collaborator on my repo when you're ready.

**Pull request closed without merging**